File: advertisingonrails/app/models/lineitem.rb

class LineItem < ApplicationRecord
  enum Status: [:Draft, :Scheduled, :Running, :Paused, :Completed, :Cancelled]
  enum PricingModel: [:CPM, :CPC, :CPA, :CPL, :CPV, :FlatFee]
  enum BidStrategy: [:Manual, :AutoMaximizeClicks, :AutoTargetCPA, :AutoTargetROAS]
  enum Pacing: [:Even, :ASAP, :Smooth]


  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  composed_of :money,
    class_name: "Money",
    mapping: [
      ${$mapping}, 
      %w[money_currency currency]
    ]

  composed_of :frequencyCap,
    class_name: "FrequencyCap",
    mapping: [
      ${$mapping}, 
      ${$mapping}, 
      %w[frequencyCap_period period]
    ]

  has_many :Campaign, class_name: 'Campaign'
  has_many :Placements, class_name: 'Placement'
  has_many :TargetingProfile, class_name: 'TargetingProfile'
  has_many :Deal, class_name: 'Deal'
  has_many :Creatives, class_name: 'CreativeAsset'
  has_many :PerformanceMetrics, class_name: 'PerformanceMetric'

end

File: advertisingonaspdotnet/Domain/LineItem.cs

using advertisingonaspdotnet.Contracts;

namespace advertisingonaspdotnet.Domain;

public class LineItem
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long? LineitemId { get; set; } 
 public virtual string? Name { get; set; } 
 public virtual Money? BidAmount { get; set; } 
 public virtual Money? DailyBudget { get; set; } 
 public virtual FrequencyCap? FrequencyCap { get; set; } 
public virtual Campaign? Campaign { get; set; } 
public virtual ICollection<Placement> Placements { get; set; } = new List<Placement>();
public virtual TargetingProfile? TargetingProfile { get; set; } 
public virtual Deal? Deal { get; set; } 
public virtual ICollection<CreativeAsset> Creatives { get; set; } = new List<CreativeAsset>();
public virtual ICollection<PerformanceMetric> PerformanceMetrics { get; set; } = new List<PerformanceMetric>();
 public virtual LineItemStatus? Status { get; set; } 
 public virtual PricingModel? PricingModel { get; set; } 
 public virtual BidStrategyType? BidStrategy { get; set; } 
 public virtual PacingType? Pacing { get; set; } 

    public static LineItem FromRequest(LineItemRequest request) {
        return new LineItem {
            Id = request.Id,
            Name = request.Name,
            BidAmount = request.BidAmount,
            DailyBudget = request.DailyBudget,
            FrequencyCap = request.FrequencyCap,
            Status = request.Status,
            PricingModel = request.PricingModel,
            BidStrategy = request.BidStrategy,
            Pacing = request.Pacing,
        };
    }
}
